server/src/concepts/authorization/controllers/auth.js

```diff
@@ -64,3 +64,21 @@ export const isThatUser = (req, res, next) => {
 
   next();
 };
+
+export const isThatUserOrAdmin = (req, res, next) => {
+  const token = req.headers.token;
+  const isCorrect = isTokenValid(token);
+
+  if (!isCorrect) {
+    return res.status(401).send("Access denied!");
+  }
+
+  const decoded = jwt.decode(token);
+  if (!decoded || !decoded.rol || decoded.sub !== req.params.id) {
```

Comment:
```suggestion
  if (!decoded || (!decoded.rol && decoded.sub !== req.params.id)) {
```
for now if **any of these three statements is false** there is no permission, because with `||` operator when something is true, code do not check further statements, so if user is not admin, we would have `you are not authorized....`. You have to wrap `!decoded.rol` and `decoded.sub !== req.params.id` with `&&` operator, because when **only both** are false we have to show this error